core/domain/translation_services_test.py

```diff
@@ -289,6 +296,139 @@ def test_compute_translation_related_change_mark_translation_needs_update(
             ], [False, True]
         )
 
+    def test_compute_translation_related_change_edits_existing_translation(
+        self
+    ) -> None:
+        translation_services.add_new_translation(
+            feconf.TranslatableEntityType.EXPLORATION, self.EXP_ID, 5, 'hi',
+            'content_5', translation_domain.TranslatedContent(
+                'Translations in Hindi!',
+                translation_domain.TranslatableContentFormat.HTML,
+                False
+            )
+        )
+        translation_services.add_new_translation(
+            feconf.TranslatableEntityType.EXPLORATION, self.EXP_ID, 5, 'hi',
+            'content_6', translation_domain.TranslatedContent(
+                'Translations in Hindi!',
+                translation_domain.TranslatableContentFormat.HTML,
+                False
+            )
+        )
+
+        entity_translation_models: Sequence[
+            translation_models.EntityTranslationsModel
+        ] = translation_models.EntityTranslationsModel.get_all().fetch()
+        self.assertEqual(len(entity_translation_models), 1)
+        entity_translation_model = entity_translation_models[0]
+        self.assertEqual(entity_translation_model.entity_version, 5)
+        self.assertEqual([
+            t['needs_update']
+            for t in entity_translation_model.translations.values()
+        ], [False, False])
+
+        self.exp.version = 6
+        change_list = [exp_domain.ExplorationChange({
+            'cmd': exp_domain.CMD_EDIT_TRANSLATION,
+            'content_id': 'content_5',
+            'language_code': 'hi',
+            'translation': translation_domain.TranslatedContent(
+                'Updated translations in Hindi!',
+                translation_domain.TranslatableContentFormat.HTML,
+                False
+            ).to_dict()
+        })]
+
+        entity_translation_models, _ = (
+            translation_services.compute_translation_related_change(
+                self.exp, change_list
+            )
+        )
```

Comment:
Nit: Add new line to differentiate between the action under test and the main assertion. Ditto below.

Reply:
Done!